### DISCUSSION TITLE
Selection limit to file picker

### DISCUSSION THREAD

**Discussion by AkshayAshokCode** [anggrayudi/SimpleStorage#112] [Q&A] [answered]
Hey!

Is it possible to add a limit to file picker selection?

**Comment by anggrayudi** [comment, marked as answer]
Currently, there's no `Intent.EXTRA_*` in Android framework to set maximum selected files. But you have 2 alternatives:

**First alternative:** Request again after the files are selected.
```kotlin
storageHelper.onFileSelected = { requestCode, files ->
    if (files.size > 4) {
        Toast.makeText(this, "Please select max 4 files", Toast.LENGTH_SHORT).show()
        storageHelper.openFilePicker(REQUEST_CODE_PICK_FILE, true)
    } else {
        // do stuff
    }
}
```

**Second alternative:** Take only as many as elements you need.
```kotlin
storageHelper.onFileSelected = { requestCode, files ->
    val selected = files.take(4)
}
```

I'll update this answer once Google added this capability via `Intent.EXTRA_*`

**Reply by AkshayAshokCode** [reply]
Yeah right, but it also compromises user experience.

What if I show a screen with the list of user selected items and tell user that only max 4 items can be selected and ask him to unselect files from this list.
And after he unselected the extra files, I'll let the user proceed. 

I want to know your views on it @anggrayudi

**Reply by anggrayudi** [reply]
Your idea has a better user experience. You can create a dialog with checkboxes, list all selected file names retrieved from `storageHelper.onFileSelected` callback, and tell the user that he should select only 4 items. Button `OK` from the dialog must be disabled if the selected files still exceed than 4 items.

**Reply by AkshayAshokCode** [reply]
Thanks @anggrayudi for the early response and also for this amazing repository! 👌🏻
